cmd/mrdegibbs.cpp

```diff
@@ -308,17 +94,24 @@ void run ()
   if (minW >= maxW)
     throw Exception ("minW must be smaller than maxW");
 
-  auto in = Image<value_type>::open (argument[0]);
-  Header header (in);
+  auto header = Header::open (argument[0]);
+  auto in = header.get_image<Degibbs::value_type>();
 
-  header.datatype() = DataType::from_command_line (DataType::Float32);
-  auto out = Image<value_type>::create (argument[1], header);
+  header.datatype() = DataType::from_command_line (header.datatype().is_complex() ? DataType::CFloat32 : DataType::Float32);
+  auto out = Image<Degibbs::value_type>::create (argument[1], header);
 
   vector<size_t> slice_axes = { 0, 1 };
   auto opt = get_options ("axes");
   const bool axes_set_manually = opt.size();
   if (opt.size()) {
     vector<uint32_t> axes = parse_ints<uint32_t> (opt[0][0]);
+    if (axes == vector<uint32_t> ({ 0, 1, 2 })) {
+      // 3D operation:
+      // process data and return immediately
+      Degibbs::unring3D (in, out, minW, maxW, nshifts);
+      return;
+    }
```

Comment:
Consider issuing a WARNING level message if the user has requested 3D operation yet the key-value field `SliceDirection` is present indicating that the acquisition was in fact 2D.

Reply:
Good idea. I think there's probably quite a lot of additional checks we can do here...